festim/exports/txt_export.py

```diff
@@ -51,83 +67,101 @@ def is_it_time_to_export(self, current_time):
                 return True
         return False
 
-    def when_is_next_time(self, current_time):
-        if self.times is None:
-            return None
-        for time in self.times:
-            if current_time < time:
-                return time
-        return None
+    def is_last(self, current_time, final_time):
+        if final_time is None:
+            # write if steady
+            return True
+        elif self.times is None:
+            if np.isclose(current_time, final_time, atol=0):
+                # write at final time if exports at each timestep
+                return True
+        else:
+            if np.isclose(current_time, self.times[-1], atol=0):
+                # write at final time if exports at specific times
+                return True
+        return False
+
+    def initialise_TXTExport(self, mesh, project_to_DG=False, materials=None):
```

Comment:
I think we should call this method something else. Maybe just `initialise` and then add docstrings to explain what it does